lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -37,6 +37,14 @@ def resources
       @resources ||= build_secrets
     end
 
+    def ejson_keys_secret
+      @ejson_keys_secret ||= begin
+        out, err, st = @kubectl.run("get", "secret", EJSON_KEYS_SECRET, output: "json")
+        out = JSON.parse(out) if st.success?
+        [out, err, st]
```

Comment:
Since the error handling is slightly different when this is called, I'm compelled to return the `out, err, st` and let the caller sort it out. Said that, calling this method `ejson_keys_secret` might not be accurate, since it encompasses more information than just the secret itself. Maybe `kubectl_get_ejson_keys`? That's more explicit but exposes implementation details 🤷‍♂️

Reply:
I think this is ok. A half thought out idea: return the secret if it's found, nil if it's not present, and raise an error for everything else, not sure if that's better.

Reply:
We can have `Kubectl` raise if not found and ignore that specifically. Here's what I would suggest: https://github.com/Shopify/kubernetes-deploy/compare/dont_prune_ejson_keys...ejson_keys_suggestion. I put it in a commit you can cherry-pick if you like it, for expediency.